resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -70,8 +70,16 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
     // If/when feature steps are executed in client mode, they should instead take care of this,
     // and this code should be removed.
     if (!sc.conf.contains(KUBERNETES_EXECUTOR_POD_NAME_PREFIX)) {
-      sc.conf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX,
-        KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name")))
+      def warnFunc(shortPrefix: String): Unit = {
+        logWarning(s"Use $shortPrefix as the executor pod's name prefix due to " +
+          s"spark.app.name is too long. Please set '${KUBERNETES_EXECUTOR_POD_NAME_PREFIX.key}' " +
+          s"if you need a custom executor pod's name prefix.")
+      }
+      val appName = sc.conf.get("spark.app.name")
+      val podNamePrefix = KubernetesConf.getResourceNamePrefix(
+        appName, KUBERNETES_POD_NAME_PREFIX_MAX_LENGTH, warnFunc)
+      assert(KubernetesUtils.isValidExecutorPodNamePrefix(podNamePrefix))
```

Comment:
I do not think this assert is needed here when `getResourceNamePrefix` is well tested.